File: android/app/src/main/java/es/manabe/yomiyasu/core/models/AuthModels.kt
package es.manabe.yomiyasu.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AuthUser(
    @SerialName("_id") val id: String,
    val username: String,
    val email: String,
    val admin: Boolean = false,
)

@Serializable
data class LoginRequest(
    val usernameOrEmail: String,
    val password: String,
    val uuid: String,
)

@Serializable
data class LoginResponse(
    val status: String = "",
    val uuid: String = "",
    val user: AuthUser,
    val accessToken: String? = null,
    val refreshToken: String? = null,
)

@Serializable
data class RefreshRequest(val uuid: String)

@Serializable
data class RefreshResponse(
    val status: String = "",
    val uuid: String = "",
    val accessToken: String? = null,
    val refreshToken: String? = null,
)

@Serializable
data class LogoutRequest(val uuid: String)

@Serializable
data class StatusResponse(val status: String = "")

File: ios/Yomiyasu/Core/Models/AuthModels.swift
import Foundation

struct AuthUser: Codable, Sendable, Equatable {
    let id: String
    let username: String
    let email: String
    let admin: Bool

    private enum CodingKeys: String, CodingKey {
        case id = "_id"
        case username
        case email
        case admin
    }
}

struct LoginRequest: Encodable, Sendable {
    let usernameOrEmail: String
    let password: String
    let uuid: String
}

struct LoginResponse: Decodable, Sendable {
    let status: String
    let uuid: String
    let user: AuthUser
    let accessToken: String?
    let refreshToken: String?
}

struct RefreshRequest: Encodable, Sendable {
    let uuid: String
}

struct RefreshResponse: Decodable, Sendable {
    let status: String
    let uuid: String
    let accessToken: String?
    let refreshToken: String?
}

struct LogoutRequest: Encodable, Sendable {
    let uuid: String
}

struct StatusResponse: Decodable, Sendable {
    let status: String
}
